### PR TITLE
Revert most of #10540, modify where `aria-selected` gets applied.

### PR DESCRIPTION
**Pull request by jansiegel**
### Context
This PR makes the following changes:

- Reverts most of #10540 
- Makes the header not get `aria-selected` when a cell/range is selected. They still get `aria-selected` when they're selected with the `navigableHeaders` option.

[skip changelog]

---
Checklist from handsontable/dev-handsontable#1576:

- [x] remove `aria-description` from `TH` root
- [x] remove `aria-label` from `<button class="changeType" />`
> [!NOTE]  
> A added an `aria-label` attribute with a space (' '), because `aria-hidden` seems to be ignored by NVDA. Without the empty `aria-label`, the button still gets read as "downpointing triangle".

- [x] add role-presentation on ` <div class="relative htLeft" role="presentation" />`
- [x] add role="presentation" on `<span class="colHeader columnSorting sortAction" />`
- [x] revert the additional div `<div class="columnSortingIndicator" aria-hidden="true"></div>`, restore `::before / ::after` logic
- [x] aria-hidden="true" on `<button class="changeType" />`
- [x] row headers shouldn't be marked as aria-selected="true" when a cell is selected, refer to https://handsoncode.atlassian.net/wiki/spaces/PROJ/pages/6297275/Accessibility+-+support+for+screen+readers

### How has this been tested?
Tested manually, modified the existing test cases.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. handsontable/dev-handsontable#1576

### Affected project(s):
- [x] `handsontable`
- [ ] `@handsontable/angular`
- [ ] `@handsontable/react`
- [ ] `@handsontable/vue`
- [ ] `@handsontable/vue3`

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)
- [ ] My change requires a change to the documentation.

### CODE REVIEW

File: handsontable/src/plugins/filters/ui/select.js [hunk 1]
```diff
@@ -124,16 +121,6 @@ export class SelectUI extends BaseUI {
...
 
-    if (this.hot.getSettings().ariaTags) {
-      setAttribute(dropdown.element, [
-        A11Y_HIDDEN()
-      ]);
-
-      setAttribute(this._element, [
-        A11Y_LISTBOX()
-      ]);
-    }
```
Comment (budnix) [inline handsontable/src/plugins/filters/ui/select.js:135]: Is this also reverted?

Comment (jansiegel) [reply handsontable/src/plugins/filters/ui/select.js:135]: Probably shouldn't be 👍

File: handsontable/src/tableView.js [hunk 2]
```diff
@@ -1298,6 +1298,10 @@ class TableView {
 
       div.className = 'relative';
       span.className = 'colHeader';
+
+      setAttribute(div, ...A11Y_PRESENTATION());
+      setAttribute(span, ...A11Y_PRESENTATION());
+
```
Comment (budnix) [inline handsontable/src/tableView.js:1304]:
```suggestion

    if (this.settings.ariaTags) {
      setAttribute(div, ...A11Y_PRESENTATION());
      setAttribute(span, ...A11Y_PRESENTATION());
    }

```
I'd be consistent with other places that add tags.

**Review by budnix — CHANGES_REQUESTED** [review]

File: handsontable/src/selection/highlight/types/focus.js [hunk 3]
```diff
@@ -7,11 +7,15 @@ import VisualSelection from '../visualSelection';
...
+ * @param {Function} highlightParams.cellAttributes Array of cell attributes.
+ * @param {Function} highlightParams.headerAttributes Array of header attributes.
  * @returns {Selection}
  */
-export function createHighlight({ cellCornerVisible, ...restOptions }) {
+export function createHighlight({ cellCornerVisible, cellAttributes, headerAttributes, ...restOptions }) {
   return new VisualSelection({
     className: 'current',
+    cellAttributes,
+    focusedHeaderAttributes: headerAttributes,
```
Comment (budnix) [inline handsontable/src/selection/highlight/types/focus.js:18]:
```suggestion
    headerAttributes: headerAttributes,
```
I'd rename the property to a more generic one. The `VisualSelection` class represents a variety of selection layers, and there should not be an option that matches the one layer - in this case "focus" layer.

File: handsontable/src/3rdparty/walkontable/src/selection/manager.js [hunk 4]
```diff
@@ -202,6 +204,16 @@ export class SelectionManager {
...
+
+          if (selectionType === 'focus' && focusedHeaderAttributes) {
+            if (!focusedHeaderAttributesMap.has(element)) {
+              focusedHeaderAttributesMap.set(element, []);
+            }
+
+            if (element.nodeName === 'TH') {
+              focusedHeaderAttributesMap.get(element).push(...focusedHeaderAttributes);
+            }
+          }
```
Comment (budnix) [inline handsontable/src/3rdparty/walkontable/src/selection/manager.js:216]:
```suggestion
          if (focusedHeaderAttributes) {
            if (!focusedHeaderAttributesMap.has(element)) {
              focusedHeaderAttributesMap.set(element, []);
            }

            if (element.nodeName === 'TH') {
              focusedHeaderAttributesMap.get(element).push(...focusedHeaderAttributes);
            }
          }
```

**Review by budnix — DISMISSED** [review]

**Review by budnix — APPROVED** [review]

**Pull request merged**